api/hypershift/v1beta1/aws.go

```diff
@@ -3,36 +3,41 @@ package v1beta1
 // AWSNodePoolPlatform specifies the configuration of a NodePool when operating
 // on AWS.
 type AWSNodePoolPlatform struct {
-	// InstanceType is an ec2 instance type for node instances (e.g. m5.large).
+	// instanceType is an ec2 instance type for node instances (e.g. m5.large).
+	// +required
+	// +kubebuilder:validation:MaxLength=255
 	InstanceType string `json:"instanceType"`
 
-	// InstanceProfile is the AWS EC2 instance profile, which is a container for an IAM role that the EC2 instance uses.
+	// instanceProfile is the AWS EC2 instance profile, which is a container for an IAM role that the EC2 instance uses.
+	// +optional
+	// +kubebuilder:validation:MaxLength=255
 	InstanceProfile string `json:"instanceProfile,omitempty"`
 
+	// subnet is the subnet to use for node instances.
 	// +kubebuilder:validation:XValidation:rule="has(self.id) && self.id.startsWith('subnet-') ? !has(self.filters) : size(self.filters) > 0", message="subnet is invalid, a valid subnet id or filters must be set, but not both"
-	// +kubebuilder:validation:Required
-	//
-	// Subnet is the subnet to use for node instances.
+	// +required
```

Comment:
So we should use `// +required` over `// +kubebuilder:validation:Required`? Did the linter make that change?

Reply:
yes, linter enforces this now.